include/deal.II/base/work_stream.h

```diff
@@ -211,6 +211,13 @@ namespace WorkStream
               , currently_in_use(in_use)
             {}
 
+            // Provide a copy constructor that actually doesn't copy the
+            // internal state. This makes handling ScratchAndCopyDataObjects
+            // easier to handle with STL containers.
+            ScratchDataObject(const ScratchDataObject &)
+              : currently_in_use(false)
+            {}
```

Comment:
I am not happy about this change (i.e., providing a "wrong" copy constructor that doesn't copy). But I grew tired of blowing up the `is_default_constructible` workaround any further.

Reply:
I think this is fine for scratch data where the state is thought of as transitory.